presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java

```diff
@@ -516,6 +517,9 @@ else if (serverConfig.isCoordinator()) {
                 .addProperty("resource_manager", String.valueOf(serverConfig.isResourceManager()))
                 .addProperty("connectorIds", nullToEmpty(serverConfig.getDataSources()));
 
+        // node monitor
+        binder.bind(ClusterSizeMonitor.class).in(Scopes.SINGLETON);
```

Comment:
It looks like we're using the `ClusterSizeMonitor` on the workers as well just because the `ServerInfoResource` is running on both the Coordinator and the Worker. This means that the logic in `ClusterSizeMonitor` will unnecessarily be running on all workers (this class adds a listener to `DiscoveryNodeManager` which will be called every time there is a change in workers/coordinators). Our use case requires the `ClusterSizeMonitor` to just be enabled on the Coordinator. I wonder if we should have a coordinator-only endpoint for this information or if we should have two versions of the `ServerInfoResource`, one for coordinator and one for worker (maybe another for the RM?).

Reply:
Rather than implementing two separate endpoints, we can add an interface for the specific part of `ClusterSizeMonitor` we're using here.  In the coordinator, we can bind that interface to `ClusterSizeMonitor`, but on the worker we can bind it to something that trivially always returns true.

Reply:
Do we see any issues if we guard ClusterSizeMonitor by checking if it's running as part of Coordinator?